plugins/destination/clickhouse/queries/tables_test.go

```diff
@@ -43,7 +43,31 @@ func TestCreateTableCluster(t *testing.T) {
 			schema.Column{Name: "extra_inet_col", Type: schema.TypeInet},
 			schema.Column{Name: "extra_inet_arr_col", Type: schema.TypeInetArray},
 		},
-	}, "my_cluster")
+	}, "my_cluster", DefaultEngine())
 
 	ensureContents(t, query, "create_table_cluster.sql")
 }
+
+func TestCreateTableWithEngine(t *testing.T) {
+	query := CreateTable(&schema.Table{
+		Name: "table_name",
+		Columns: schema.ColumnList{
+			schema.CqIDColumn,
+			schema.CqParentIDColumn,
+			schema.CqSourceNameColumn,
+			schema.CqSyncTimeColumn,
+			schema.Column{
+				Name:            "extra_col",
+				Type:            schema.TypeFloat,
+				CreationOptions: schema.ColumnCreationOptions{NotNull: true},
+			},
+			schema.Column{Name: "extra_inet_col", Type: schema.TypeInet},
+			schema.Column{Name: "extra_inet_arr_col", Type: schema.TypeInetArray},
+		},
+	}, "", &Engine{
+		Name:       "ReplicatedMergeTree",
+		Parameters: []any{"a", "b", 1, 2, 3},
```

Comment:
Should we maybe test some of the other cases as well, like float and bool? Or are they being tested elsewhere?

Reply:
No, didn't add tests for those. I'll add them soon

Reply:
done!